curp/src/client_new/unary.rs

```diff
@@ -0,0 +1,620 @@
+                let _ig = self.connects.insert(diff, new_conn);
+            }
+        }
+        let sames = &old_ids & &new_ids;
+        for same in sames {
+            let conn = self
+                .connects
+                .get(&same)
+                .unwrap_or_else(|| unreachable!("{same} must in old connects"));
+            let addrs = member_addrs
+                .get(&same)
+                .unwrap_or_else(|| unreachable!("{same} must in new member addrs"))
+                .clone();
+            conn.update_addrs(addrs).await?;
+        }
+        Ok(())
+    }
+}
+
+/// The unary client config
+#[derive(Debug)]
+pub(super) struct UnaryConfig {
+    /// The rpc timeout of a propose request
+    propose_timeout: Duration,
+    /// The rpc timeout of a 2-RTT request, usually takes longer than propose timeout
+    /// Default to 2 * propose_timeout, The recommended the values is within
+    /// (propose_timeout, 2 * propose_timeout].
+    wait_synced_timeout: Duration,
+}
+
+impl UnaryConfig {
+    /// Create a unary config
+    fn new(propose_timeout: Duration) -> Self {
+        Self {
+            propose_timeout,
+            wait_synced_timeout: propose_timeout * 2,
+        }
+    }
+
+    /// Create a unary config
+    fn new_full(propose_timeout: Duration, wait_synced_timeout: Duration) -> Self {
+        Self {
+            propose_timeout,
+            wait_synced_timeout,
+        }
+    }
+}
+
+/// The unary client
+#[derive(Debug)]
+pub(super) struct Unary<C: Command> {
+    /// Cluster state
+    cluster_state: ClusterState,
+    /// Leader state
+    leader_state: RwLock<LeaderState>,
+    /// Local server id
+    local_server_id: Option<ServerId>,
+    /// Unary config
+    config: UnaryConfig,
+    /// marker
+    phantom: PhantomData<C>,
+}
+
+impl<C: Command> Unary<C> {
+    /// Get cluster version.
+    fn cluster_version(&self) -> u64 {
+        self.cluster_state
+            .version
+            .load(std::sync::atomic::Ordering::Relaxed)
+    }
+
+    /// Bypass a server by providing a Protocol implementation, see [`crate::server::Rpc`].
+    fn bypass<P: Protocol>(&self, id: ServerId, server: P) {
+        let _ig = self
+            .cluster_state
+            .connects
+            .insert(id, Arc::new(BypassedConnect::new(id, server)));
+    }
+
+    /// Give a handle `f` to apply to all servers *concurrently* and return a stream to poll result one by one.
+    fn for_each_server<R, F: Future<Output = R>>(
+        &self,
+        f: impl FnMut(Arc<dyn ConnectApi>) -> F,
+    ) -> (usize, impl Stream<Item = R>) {
+        let connects = self
+            .cluster_state
+            .connects
+            .iter()
+            .map(|connect| Arc::clone(&connect))
+            .collect_vec();
+        // quorum calculated here to keep quorum = stream.len(), otherwise Non-atomic read operation on the `connects` may result in inconsistency.
+        let quorum = connects.len();
```

Comment:
Why this is called `quorum`, shouldn't it be called something like `cluster_size`?

Reply:
I feel like they are the same meaning.

`super quorum` : 3/4 * cluster size -> `super`: 3/4, `quorum`: cluster size

:D

Reply:
The reason why we call it super quorum is the super quorum (3/4 * cluster_size) is greater than  the quorum (1/2 * (cluster_size + 1)). It should be cluster_size, not quorum here.

Reply:
OK, I'll correct it.